packages/core/src/project-creation.ts

```diff
@@ -0,0 +1,125 @@
+import path from 'node:path'
+import fs from 'node:fs'
+import os from 'node:os'
+import { getRecommendedGitIgnore, getRecommendedPackageJson } from './project-structure'
+import { getPackageRoot, getPackageJson } from './util/packageInfo'
+import { prompt } from 'enquirer'
+import { merge } from 'lodash'
+import chalk from 'chalk'
+
+function printAsciiLogo() {
+  console.info(chalk.blue(`888    d8P                    d8b `))
+  console.info(chalk.blue(`888   d8P                     Y8P `))
+  console.info(chalk.blue(`888  d8P                          `))
+  console.info(chalk.blue(`888d88K     888  888  8888b.  888 `))
+  console.info(chalk.blue(`8888888b    888  888     "88b 888 `))
+  console.info(chalk.blue(`888  Y88b   888  888 .d888888 888 `))
+  console.info(chalk.blue(`888   Y88b  Y88b 888 888  888 888 `))
+  console.info(chalk.blue(`888    Y88b  "Y88888 "Y888888 888 `))
+  console.info('')
+}
+
+async function printWelcomeMessage() {
+  const packageJson = await getPackageJson()
+
+  console.info(chalk.cyan(`Welcome to kuai v${packageJson.version}\n`))
+}
+
+async function copySampleProject(projectRoot: string) {
+  const packageRoot = getPackageRoot()
+
+  fs.existsSync(projectRoot)
+
+  const sampleProjectPath = path.join(packageRoot, 'sample-projects')
+
+  const sampleProjectRootFiles = fs.readdirSync(sampleProjectPath)
+  const existingFiles = sampleProjectRootFiles
+    .map((f) => path.join(projectRoot, f))
+    .filter((f) => fs.existsSync(f))
+    .map((f) => path.relative(process.cwd(), f))
+
+  if (existingFiles.length > 0) {
+    const errorMsg = `We couldn't initialize the sample project because these files already exist: ${existingFiles.join(
+      ', ',
+    )}
+
+Please delete or move them and try again.`
+    console.error(errorMsg)
+    process.exit(1)
+  }
+
+  fs.cpSync(path.join(packageRoot, 'sample-projects'), projectRoot, { recursive: true })
+}
+
+async function addGitIgnore(projectRoot: string) {
+  const gitIgnorePath = path.join(projectRoot, '.gitignore')
+
+  const content = await getRecommendedGitIgnore()
+
+  if (fs.existsSync(gitIgnorePath)) {
+    fs.appendFileSync(gitIgnorePath, content)
+  }
```

Comment:
`return` is required in this branch, or `fs.writeFileSync` will be called after `fs.appendFileSync`